internal/rbd/rbd_util.go

```diff
@@ -881,6 +867,32 @@ func genVolFromVolID(ctx context.Context, volumeID string, cr *util.Credentials,
 	return rbdVol, err
 }
 
+// genVolFromVolID generates a rbdVolume structure from the provided identifier, updating
+// the structure with elements from on-disk image metadata as well.
+func genVolFromVolID(ctx context.Context, volumeID string, cr *util.Credentials, secrets map[string]string) (*rbdVolume, error) {
+	vol, err := generateVolumeFromVolumeID(ctx, volumeID, cr, secrets)
+	if !errors.Is(err, util.ErrKeyNotFound) && !errors.Is(err, util.ErrPoolNotFound) {
+		return vol, err
+	}
+	c := util.NewK8sClient()
+	listOpt := metav1.ListOptions{
+		LabelSelector: PVReplicatedLabelKey,
+	}
+	pvlist, pErr := c.CoreV1().PersistentVolumes().List(context.TODO(), listOpt)
+	if pErr != nil {
+		return vol, pErr
+	}
+	for i := range pvlist.Items {
+		if pvlist.Items[i].Spec.CSI != nil && pvlist.Items[i].Spec.CSI.VolumeHandle == volumeID {
+			if v, ok := pvlist.Items[i].Annotations[PVVolumeHandleAnnotationKey]; ok {
+				util.UsefulLog(ctx, "found new volumeID %s for existing volumeID %s", v, volumeID)
+				return generateVolumeFromVolumeID(ctx, v, cr, secrets)
```

Comment:
@Madhu-1 this is bit confusing. That said, Is it a problem if we see new volume ID for existing Volume ID?  can we add some source code comment about this block logic ?

Reply:
@humblec it's not a problem.